test/core/memory_grow.wast

```diff
@@ -0,0 +1,34 @@
+;; Test that newly allocated memory (program start and memory.grow) is zeroed
+
+(module
+  (memory 1)
+  (func (export "grow") (param i32) (result i32)
+    (memory.grow (get_local 0))
+  )
+  (func (export "check-memory-zero") (param i32 i32) (result i32)
+    (local i32)
+    (set_local 2 (i32.const 1))
+    (block
+      (loop
+        (set_local 2 (i32.load8_u (get_local 0)))
+        (br_if 1 (i32.ne (get_local 2) (i32.const 0)))
+        (br_if 1 (i32.ge_u (get_local 0) (get_local 1)))
+        (set_local 0 (i32.add (get_local 0) (i32.const 1)))
+        (br_if 0 (i32.le_u (get_local 0) (get_local 1)))
+      )
+    )
+    (get_local 2)
+  )
+)
+
+(assert_return (invoke "check-memory-zero" (i32.const 0) (i32.const 0xffff)) (i32.const 0))
+(assert_return (invoke "grow" (i32.const 1)) (i32.const 1))
+(assert_return (invoke "check-memory-zero" (i32.const 0) (i32.const 0x1_ffff)) (i32.const 0))
```

Comment:
Nit: this could use 0x10000 as lower bound to just check the new bytes; similarly the ones below.